lib/capistrano/tasks/bundler.cap

```diff
@@ -28,7 +28,8 @@ namespace :bundler do
           options = []
           options << "--gemfile #{fetch(:bundle_gemfile)}" if fetch(:bundle_gemfile)
           options << "--path #{fetch(:bundle_path)}" if fetch(:bundle_path)
-          if test(:bundle, :check, *options)
+
+          unless fetch(:bundle_check_before_install) && test(:bundle, :check, *options)
```

Comment:
I'm not sure this is correct…
```ruby
unless fetch(:bundle_check_before_install) && test(:bundle, :check, *options)
```
which is the same as:
```ruby
if !(fetch(:bundle_check_before_install) && test(:bundle, :check, *options))
```
Which (applying DeMorgan's laws) is the same as
```ruby
if !fetch(:bundle_check_before_install) || !test(:bundle, :check, *options)
```

So if the `bundle_check_before_install` flag is false or `bundle check` returns false, we decide the dependencies are satisfied and skip installation.

Should it be:

```suggestion
          if fetch(:bundle_check_before_install) && test(:bundle, :check, *options)
```

where if the flag is true and bundle check returns true, we skip installation. If either of those are false, we run the installation.

Am I reading this correctly?

Reply:
I think that would be correct. I apologize, I haven't messed with this in quite a while so I am a bit rusty on all of the details.